src-tauri/src/telemetry_manager.rs

```diff
@@ -0,0 +1,321 @@
+use anyhow::Result;
+use jsonwebtoken::errors::Error as JwtError;
+use jsonwebtoken::{decode, Algorithm, DecodingKey, TokenData, Validation};
+use log::{debug, error, info, warn};
+use serde::{Deserialize, Serialize};
+use std::{sync::Arc, thread::sleep, time::Duration};
+use tari_common::configuration::Network;
+use tari_core::transactions::tari_amount::MicroMinotari;
+use tokio::sync::RwLock;
+use tokio_util::sync::CancellationToken;
+
+use crate::{
+    app_config::{AppConfig, MiningMode},
+    cpu_miner::CpuMiner,
+    gpu_miner::GpuMiner,
+    hardware_monitor::HardwareMonitor,
+    node_manager::NodeManager,
+    UniverseAppState, LOG_TARGET,
+};
+
+#[derive(Debug, Deserialize, Serialize)]
+struct AirdropAccessToken {
+    exp: u64,
+    iat: i32,
+    id: String,
+    provider: String,
+    role: String,
+    scope: String,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum TelemetryResource {
+    Cpu,
+    Gpu,
+    #[serde(rename(serialize = "cpu-gpu"))]
+    CpuGpu,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub enum TelemetryNetwork {
```

Comment:
nit: Why not use the full name and just use `serde(lowercase)`?

Reply:
It tried it but it failed at lowercasing the variants... not sure why